### PR TITLE
Introduce PUBLIC_CLOUD_EC2_BOOT_MODE

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#128459](https://progress.opensuse.org/issues/128459)
- Verification run:

### CODE REVIEW

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: lib/publiccloud/ec2.pm [hunk 1]
```diff
@@ -156,6 +157,7 @@ sub upload_img {
           . "--ec2-ami '" . $helper_ami_id . "' "
           . "--type '" . $instance_type . "' "
           . "--user '" . $self->provider_client->username . "' "
+          . "--boot-mode '" . $boot_mode . "' "
```
Comment (grisu48) [inline lib/publiccloud/ec2.pm:160]: Wouldn't it be better to not set this at all, if the `PUBLIC_CLOUD_EC2_BOOT_MODE` setting is not set?

Comment (grisu48) [reply lib/publiccloud/ec2.pm:160]: ok, fine by me!

**Review by grisu48 — APPROVED** [review]

**Review by ricardobranco777 — APPROVED** [review]
LGTM

**Comment by asmorodskyi** [comment]
code is fine but pipeline is broken please fix it

**Comment by Martchus** [comment]
The pipeline has already been fixed on master. So rebasing should help.

**Comment by pdostal** [comment]
Thank you @Martchus

**Pull request merged**